azuredevops/internal/service/servicehook/resource_servicehook_storage_queue.go

```diff
@@ -0,0 +1,192 @@
+		Type:         schema.TypeString,
+		Required:     true,
+		Sensitive:    true,
+		ValidateFunc: validation.StringLenBetween(64, 100),
+	}
+	resourceSchema["queue_name"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Required: true,
+	}
+	resourceSchema["visi_timeout"] = &schema.Schema{
+		Type:     schema.TypeInt,
+		Optional: true,
+		Default:  "0",
+	}
+	resourceSchema["ttl"] = &schema.Schema{
+		Type:     schema.TypeInt,
+		Optional: true,
+		Default:  "604800",
+	}
+
+	return &schema.Resource{
+		Create: resourceServicehookStorageQueueCreate,
+		Read:   resourceServicehookStorageQueueRead,
+		Update: resourceServicehookStorageQueueUpdate,
+		Delete: resourceServicehookStorageQueueDelete,
+
+		Schema: resourceSchema,
+	}
+}
+
+func resourceServicehookStorageQueueCreate(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+	subscription, err := expandServicehookStorageQueue(d)
+	if err != nil {
+		return err
+	}
+
+	createdSubscription, err := createSubscription(d, clients, subscription)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(createdSubscription.Id.String())
+	return resourceServicehookStorageQueueRead(d, m)
+}
+
+func resourceServicehookStorageQueueRead(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+	subscriptionId := converter.UUID(d.Id())
+	subscription, err := getSubscription(clients, subscriptionId)
+	if err != nil {
+		return err
+	}
+	flattenServicehookStorageQueue(d, subscription, d.Get("account_key").(string))
+	return nil
+}
+
+func resourceServicehookStorageQueueUpdate(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+	subscription, err := expandServicehookStorageQueue(d)
+	if err != nil {
+		return err
+	}
+
+	newSubscription, err := updateSubscription(clients, subscription)
+	if err != nil {
+		return err
+	}
+
+	flattenServicehookStorageQueue(d, newSubscription, d.Get("account_key").(string))
+	return resourceServicehookStorageQueueRead(d, m)
+}
+
+func resourceServicehookStorageQueueDelete(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*client.AggregatedClient)
+
+	return clients.ServiceHooksClient.DeleteSubscription(clients.Ctx, servicehooks.DeleteSubscriptionArgs{
+		SubscriptionId: converter.UUID(d.Id()),
+	})
+}
+
+func expandServicehookStorageQueue(d *schema.ResourceData) (*servicehooks.Subscription, error) {
+	var subscriptionId *uuid.UUID
+	parsedID, err := uuid.Parse(d.Id())
+	if err == nil {
+		subscriptionId = &parsedID
+	}
+	visiTimeout := strconv.Itoa(d.Get("visi_timeout").(int))
+	ttl := strconv.Itoa(d.Get("ttl").(int))
+	publisherInputs := expandPublisherInputs(d.Get("project_id").(string), d.Get("publisher").([]interface{}))
+	resourceVersion := publisherResourceVersionMap[d.Get("publisher.0.name").(string)]
+	return &servicehooks.Subscription{
+		Id:               subscriptionId,
+		ConsumerActionId: &consumerActionId,
+		ConsumerId:       &consumerId,
```

Comment:
Since this resource stick to storage queue resource, the consumer ID can be hard coded. Currently there is only one action ID available, which can also be hardcoded.